packages/terra-application-header-layout/src/ApplicationHeaderLayout.jsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import AppDelegate from 'terra-app-delegate';
+
+import styles from './ApplicationHeaderLayout.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The AppDelegate instance provided by the containing component. If present, its properties will propagate to the children components.
+   * */
+  app: AppDelegate.propType,
+  /**
+   * Extensions element to be placed before the end of the header.
+   * */
+  extensions: PropTypes.element,
+  /**
+   * Logo element to be placed at the start of the header after the toggle element.
+   * */
+  logo: PropTypes.element,
+  /**
+   * Layout configuration to handle size and toggle functionality.
+   * */
+  layoutConfig: PropTypes.shape({
+    size: PropTypes.string,
+    toggleMenu: PropTypes.func,
+    menuIsOpen: PropTypes.bool,
+  }),
+  /**
+   * Navigation element to be placed within the fill area of the header.
+   * */
+  navigation: PropTypes.element,
+  /**
+   * Toggle element to be placed at the start of the header.
+   * */
+  toggle: PropTypes.element,
+  /**
+   * Utilities element to be placed at the end of the header.
+   * */
+  utilities: PropTypes.element,
+};
+
+const defaultProps = {
+  layoutConfig: {
+    menuIsOpen: false,
+  },
+};
+
+const appendSizeToElement = (app, size, element) => React.cloneElement(element, { app, size });
+const appendConfigToElement = (app, layoutConfig, element) => React.cloneElement(element, { app, layoutConfig });
```

Comment:
Can we document somewhere that the size and layout config are getting added to elements.